translate/public/locale/en-US/translate.ftl

```diff
@@ -440,8 +440,7 @@ interactivetour-InteractiveTour--end-footer =
 
 ## Machinery
 ## Shows a list of translations from machines.
-machinery-Machinery--search-placeholder =
-    .placeholder = Concordance Search
+machinery-Machinery--search-placeholder = Concordance Search
```

Comment:
The machinery search state management is challenging because the input and the query are not in sync, and the results for an empty string are qualitatively different.

The implementation proposed here has the following data fields:
- `input` is what's currently displayed in the text box
- `query` is a submitted `input`, which corresponds to the `results`
- `results` are fetched & displayed async whenever `query` updates

Based on these, we have the following states:
1. empty `input`, empty `query`: placeholder "Concordance Search", icon 🔍 does nothing
2. non-empty `input`, `input` ≠ `query`: show `input`, icon 🔍 executes search
3. non-empty `input`, `input` = `query`: show `input`, icon ✖️ resets to initial state (1)
4. empty `input`, non-empty `query`: placeholder `query`, icon ✖️ resets to initial state (1)

If I've understood right, your concern is about the definition and behaviour of state 4 above, yes? I'd be happy to redefine that according to your specification.

The current solution is also the reason for the `.ftl` file change, as the placeholder message now needs to be fetched a bit differently, and that's much easier if the message is in a Fluent value rather than an attribute: https://github.com/mozilla/pontoon/blob/38bff45c44d300b4978f2e2cd1ec7c2f771c27d3/translate/src/modules/machinery/components/Machinery.tsx#L32-L38

Reply:
> If I've understood right, your concern is about the definition and behaviour of state 4 above, yes? I'd be happy to redefine that according to your specification.

Correct. Placeholder should in this case reset to "Concordance Search".

Reply:
Placeholder now always set as "Concordance Search".